src/transformers/modeling_utils.py

```diff
@@ -3207,9 +3208,39 @@ def from_pretrained(
                         )
                         if resolved_archive_file is not None:
                             is_sharded = True
-                    if resolved_archive_file is None:
-                        # Otherwise, maybe there is a TF or Flax model file.  We try those to give a helpful error
-                        # message.
+
+                    if resolved_archive_file is not None:
+                        if filename in [WEIGHTS_NAME, WEIGHTS_INDEX_NAME]:
```

Comment:
one thing I would be wary is just that if we convert a big checkpoint from torch to safetensors and we want to load it in `Flax`, sharded safetensors are not supported yet

Reply:
Flax defaults to loading flax checkpoints, not safetensors, so it won't be affected by a repo where there is sharded safetensors